data.sql

```diff
@@ -1,13 +1,83 @@
-/* Populate database with sample data. */
 -- data.sql
-USE vet_clinic;
 
-INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES
-('Agumon', '2020-02-03', 0, TRUE, 10.23),
-('Gabumon', '2018-11-15', 2, TRUE, 8),
-('Pikachu', '2021-01-07', 1, FALSE, 15.04),
-('Devimon', '2017-05-12', 5, TRUE, 11);
+-- INSERT animals data
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
+VALUES ('Agumon', '2020-02-03', 0, TRUE, 10.23, NULL);
 
-INSERT INTO animals (name) VALUES ('Luna');
-INSERT INTO animals (name) VALUES ('Daisy');
-INSERT INTO animals (name) VALUES ('Charlie');
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
+VALUES ('Gabumon', '2018-11-15', 2, TRUE, 8, NULL);
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
+VALUES ('Pikachu', '2021-01-07', 1, FALSE, 15.04, NULL);
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
+VALUES ('Devimon', '2017-05-12', 5, TRUE, 11, NULL);
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
+VALUES ('Charmander', '2020-02-08', 0, FALSE, -11, NULL);
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
+VALUES ('Plantmon', '2021-11-15', 2, TRUE, -5.7, NULL);
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
+VALUES ('Squirtle', '1993-04-02', 3, FALSE, -12.13, NULL);
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
+VALUES ('Angemon', '2005-06-12', 1, TRUE, -45, NULL);
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
+VALUES ('Boarmon', '2005-06-07', 7, TRUE, 20.4, NULL);
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
+VALUES ('Blossom', '1998-10-13', 3, TRUE, 17, NULL);
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
+VALUES ('Ditto', '2022-05-14', 4, TRUE, 22, NULL);
+
+-- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
+BEGIN;
+
+UPDATE animals SET species = 'unspecified';
+
+-- Verify the change (optional)
+SELECT * FROM animals;
+
+ROLLBACK;
+
+-- Verify that the species column went back to the original state (optional)
+SELECT * FROM animals;
+
+-- Inside a transaction update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
+-- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+-- Verify that changes were made.
+COMMIT;
```

Comment:
- [ ] Since you used rollback you don't have to use commit here, please remove it.

Reply:
done